maven-plugin/src/main/java/org/wildfly/galleon/maven/UserFeaturePackBuildMojo.java

```diff
@@ -147,8 +149,25 @@ protected void doExecute() throws MojoExecutionException, MojoFailureException {
     private void addModulePackages(final Path srcModulesDir, final FeaturePackDescription.Builder fpBuilder, final Path targetResources) throws MojoExecutionException {
         final Map<String, Path> moduleXmlByPkgName = new HashMap<>();
         try {
-            Util.findModules(srcModulesDir, moduleXmlByPkgName);
-            packageModules(fpBuilder, targetResources, moduleXmlByPkgName, null);
+            final Path addOnsDir = srcModulesDir.resolve(WfConstants.SYSTEM).resolve(WfConstants.ADD_ONS);
+            if (Files.exists(addOnsDir)) {
```

Comment:
Now if the addons are there we are not going to be looking for modules in the original modules dir?

Reply:
@aloubyansky , I made an invalid shortcut there, we could have modules under module, system/add-ons and even system/layers. So the 3 locations should be handled. I am reviewing that.